desc/objectives/_wrappers.py

```diff
@@ -139,7 +141,13 @@ def _update_equilibrium(self, x, store=False):
         After updating, if store=False, self._eq will revert back to the previous
         solution when store was True
         """
-        if jnp.allclose(x, self._x_old, rtol=1e-14, atol=1e-14):
+        from desc.optimize.utils import f_where_x
+
+        # first check if its something we've seen before, if it is just return
```

Comment:
is this is where we basically can see if we've already seen the same x, then the optimizer did not accept the step and we dont need to do anything?

Reply:
the more common case is that we saw this same `x` when we tried this step and evaluated `f` but haven't actually moved to the new x yet. After the step is accepted, we want to compute the jacobian, but we don't want to re-solve the equilibrium because we already solved it when trying that step the first time